subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/FileSystemSnapshotBuilderTest.groovy

```diff
@@ -123,7 +118,8 @@ class FileSystemSnapshotBuilderTest extends Specification {
         def result = builder.build()
 
         then:
-        result == snapshot
+        result instanceof RegularFileSnapshot
+        result.hash == hasher.hash(new File(basePath), snapshot)
```

Comment:
Why did we need this change?

Reply:
`snapshot` is not a `RegularFileSnapshot` any more. The `RegularFileSnapshot` is now created by the `addFile` method. So I can only assert some properties of the result.